react/features/base/avatar/components/native/RemoteAvatar.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+
+import React, { PureComponent } from 'react';
+import FastImage from 'react-native-fast-image';
+
+import styles from './styles';
+
+type Props = {
+
+    /**
+     * Callback for load errors.
+     */
+    onError: Function,
+
+    /**
+     * Size of the avatar.
+     */
+    size: number,
+
+    /**
+     * URI of the avatar to load.
+     */
+    uri: ?string
+};
+
+/**
+ * Implements a private class that is used to fetch and render remote avatars based on an URI.
+ */
+export default class RemoteAvatar extends PureComponent<Props> {
```

Comment:
random: we should really re-evaluate using functional components...

Reply:
I agree